bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVeneerSettingsFactory.java

```diff
@@ -0,0 +1,327 @@
+    RetrySettings retrySettings =
+        buildIdempotentRetrySettings(builder.sampleRowKeysSettings().getRetrySettings(), options);
+
+    builder.sampleRowKeysSettings()
+        .setRetrySettings(retrySettings)
+        .setRetryableCodes(buildRetryCodes(options.getRetryOptions()));
+  }
+
+  /** To build BigtableDataSettings#mutateRowSettings with default Retry settings. */
+  private static void buildMutateRowSettings(Builder builder, BigtableOptions options) {
+    RetrySettings retrySettings =
+        buildIdempotentRetrySettings(builder.mutateRowSettings().getRetrySettings(), options);
+
+    builder.mutateRowSettings()
+        .setRetrySettings(retrySettings)
+        .setRetryableCodes(buildRetryCodes(options.getRetryOptions()));
+  }
+
+  /** To build default Retry settings for Point Read. */
+  private static void buildReadRowSettings(Builder builder, BigtableOptions options) {
+    RetrySettings retrySettings =
+        buildIdempotentRetrySettings(builder.readRowSettings().getRetrySettings(), options);
+
+    builder.readRowSettings()
+        .setRetrySettings(retrySettings)
+        .setRetryableCodes(buildRetryCodes(options.getRetryOptions()));
+  }
+
+  /** To build BigtableDataSettings#readRowsSettings with default Retry settings. */
+  private static void buildReadRowsSettings(Builder builder, BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+
+    RetrySettings.Builder retryBuilder = builder.readRowsSettings().getRetrySettings().toBuilder()
+        .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+        .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()))
+        .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries());
+
+    // configurations for RPC timeouts
+    Duration readPartialRowTimeout = ofMillis(retryOptions.getReadPartialRowTimeoutMillis());
+    retryBuilder
+        .setInitialRpcTimeout(readPartialRowTimeout)
+        .setMaxRpcTimeout(readPartialRowTimeout)
+        .setTotalTimeout(ofMillis(options.getCallOptionsConfig().getLongRpcTimeoutMs()));
+
+    builder.readRowsSettings()
+        .setRetrySettings(retryBuilder.build())
+        .setRetryableCodes(buildRetryCodes(options.getRetryOptions()));
+  }
+
+  /** Creates {@link RetrySettings} for non-streaming idempotent method. */
+  private static RetrySettings buildIdempotentRetrySettings(RetrySettings retrySettings,
+      BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+    RetrySettings.Builder retryBuilder = retrySettings.toBuilder();
+
+    retryBuilder
+        .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+        .setMaxRetryDelay(ofMillis(retryOptions.getMaxElapsedBackoffMillis()))
+        .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries());
```

Comment:
There is no max attempts in the classic client. Please use 0 to disable

Reply:
removed overriding of maxAttempts(except ReadRows), it would now be defaults to `setMaxAttempts(0)`.